### PR TITLE
[PATCH v2] Classifier shm usage optimization

### PR DESCRIPTION
**Pull request by psavol**
Combine three SHM blocks into one. Modified sysinfo example to print out shm blocks, so that it's easy to see how many blocks ODP uses internally. This is how I found out inefficient SHM usage of CLS.

### CODE REVIEW

File: platform/linux-generic/odp_classification.c [hunk 1]
```diff
@@ -80,102 +90,46 @@ pmr_t *get_pmr_entry_internal(odp_pmr_t pmr)
...
-	odp_shm_t queue_grp_shm;
+	odp_shm_t shm;
 	int i;
 
-	cos_shm = odp_shm_reserve("_odp_shm_odp_cos_tbl",
-				  sizeof(cos_tbl_t),
-				  sizeof(cos_t),
-				  _ODP_SHM_NO_HP);
+	shm = odp_shm_reserve("_odp_cls_global", sizeof(cls_global_t),
+			      ODP_CACHE_LINE_SIZE, 0);
```
Comment (muvarov) [inline platform/linux-generic/odp_classification.c]: that is small mapping, why did you remove _ODP_SHM_NO_HP here?

Comment (psavol) [reply platform/linux-generic/odp_classification.c]: Matias will send shortly PR which removes/replaces the flag and changes HP mapping. We did run into latency problems with the current NO_HP mappings. The PR will suggest to use normal pages for <64kB mappings, and HP for larger ones.

Also another aspect to this is number of page table entries used. The smaller the pages size the more page table entries are used, and it's more likely to run into TLB misses (which hurt performance and latency). 

This struct is about 150kB (38x4kB pages vs one 2MB page).

**Comment by muvarov** [comment]
<pre>v1 checks:
checkpatch.pl - OK
ODP License Agreement - PASSED
</pre>

**Comment by psavol** [comment]
v2: rebased

**Comment by muvarov** [comment]
Merged.
